src/openrct2-ui/windows/EditorInventionsList.cpp

```diff
@@ -604,8 +604,8 @@ class InventionDragWindow final : public Window
             if (!inventionListWindow->IsResearchItemSelected(research))
             {
                 inventionListWindow->SetSelectedResearchItem(research);
-                inventionListWindow->Invalidate();
             }
+            inventionListWindow->Invalidate();
```

Comment:
Looking at this again, I don't think this change is actually the cause of this bug, but rather a side effect. It just happens to work for when you're dragging on the window, since that invalidates the same area. The real issue is the width of the drag-window that is not calculated correctly. Strings with 2 parts are too small, so the text renders outside the "window", leaving things behind that are outside its invalidation area.

Reply:
https://github.com/OpenRCT2/OpenRCT2/blob/2681443b589664c9a29e084f333c3d774e56f028/src/openrct2-ui/windows/EditorInventionsList.cpp#L667

Reply:
Looks like you only need to replace the string Id of the formatter, since the one it currently uses has no vehicle part. `STR_INVENTIONS_LIST_RIDE_AND_VEHICLE_NAME_DRAG` is the one it needs.

Reply:
Gotcha, I will try and post the result

Reply:
![2](https://user-images.githubusercontent.com/12020398/194687753-7212cbc9-af3e-4e22-b979-df7bf49e9267.gif)

@Broxzier Sadly, it was also not a full fix of this bug. Once the text window exceeds scrollbox boundary, it still makes leftover outside.